linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/account_password_pam_faillock_password_auth/oval/shared.xml

```diff
@@ -0,0 +1,24 @@
+<def-group>
+    <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata(
+        "Configure the use of the pam_faillock.so module in the /etc/pam.d/password-auth file.")
+    }}}
+        <criteria>
+            <criterion test_ref="test_{{{ rule_id }}}"
+                    comment="The faillock module is present in /etc/pam.d/password-auth"/>
+        </criteria>
+    </definition>
+    <ind:textfilecontent54_test check="all" check_existence="at_least_one_exists" version="1"
+    id="test_{{{ rule_id }}}"
+    comment="Check the presence of pam_faillock.so in password-auth">
+        <ind:object
+        object_ref="obj_{{{ rule_id }}}"/>
+    </ind:textfilecontent54_test>
+    <ind:textfilecontent54_object id="obj_{{{ rule_id }}}"
+    comment="Obtain pam_faillock.so ocurrences in password-auth" version="1">
+        <ind:filepath >/etc/pam.d/password-auth</ind:filepath>
+        <!-- pam_faillock.so needs to be before pam_unix\.so -->
+        <ind:pattern operation="pattern match">\s*auth\s*\S+\s*pam_faillock\.so\s*preauth[\s\S]*?pam_unix\.so</ind:pattern>
```

Comment:
pam_faillock.so is implemented by two lines (preauth and authfail) in auth section and one line in account section. The OVAL should also check this to avoid passing incomplete settings. You can use the [OVAL](https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_deny/oval/shared.xml) in `accounts_passwords_pam_faillock_deny` or another rule related to pam_faillock for reference.